### PR TITLE
[face_recognition] add nodelet / script / message files for face recognition (new)

### PR DESCRIPTION
**Pull request by furushchev**
This pull request overrides #41.
The differences from #41 are
- Support original face origin (because #39 breaks api)
- Track OpenCV 3.1 / 3.2 issue of face_detection

face_recognition

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -63,7 +64,6 @@ add_message_files(
    FlowStamped.msg
    FlowArray.msg
    FlowArrayStamped.msg
-   RectArrayStamped.msg
```
Comment (k-okada) [inline CMakeLists.txt:66]: do not remove this

File: cfg/FaceRecognition.cfg [hunk 2]
```diff
@@ -0,0 +1,59 @@
+#! /usr/bin/env python
+# Software License Agreement (BSD License)
+#
+# Copyright (c) 2014, Kei Okada.
```
Comment (k-okada) [inline cfg/FaceRecognition.cfg]: Please update with your name

File: scripts/face_recognition_trainer.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
+#! /usr/bin/env python
+# Software License Agreement (BSD License)
+#
+# Copyright (c) 2014, Kei Okada.
```
Comment (k-okada) [inline scripts/face_recognition_trainer.py]: please update

File: src/nodelet/face_recognition_nodelet.cpp [hunk 4]
```diff
@@ -0,0 +1,545 @@
+/*********************************************************************
+* Software License Agreement (BSD License)
+*
+*  Copyright (c) 2014, Kei Okada.
```
Comment (k-okada) [inline src/nodelet/face_recognition_nodelet.cpp]: update

File: CMakeLists.txt (hunk 1 above)
Comment (furushchev) [reply CMakeLists.txt:66]: Why? You duplicate this definition on purpose?

https://github.com/ros-perception/opencv_apps/blob/indigo/CMakeLists.txt#L61
https://github.com/ros-perception/opencv_apps/blob/indigo/CMakeLists.txt#L66

File: cfg/FaceRecognition.cfg (hunk 2 above)
Comment (furushchev) [reply cfg/FaceRecognition.cfg]: I'm very sorry... Updated.

File: scripts/face_recognition_trainer.py (hunk 3 above)
Comment (furushchev) [reply scripts/face_recognition_trainer.py]: Updated.

File: src/nodelet/face_recognition_nodelet.cpp (hunk 4 above)
Comment (furushchev) [reply src/nodelet/face_recognition_nodelet.cpp]: Updated.

File: CMakeLists.txt (hunk 1 above)
Comment (furushchev) [reply CMakeLists.txt:66]: @k-okada kindly ping

**Comment by k-okada** [comment]
ok, so
1. what is `Track OpenCV 3.1 / 3.2 issue of face_detection`
2. could you make better images which wen can used in wiki.ros.org?

**Comment by furushchev** [comment]
@k-okada Thank you for quick response!

> what is Track OpenCV 3.1 / 3.2 issue of face_detection

Oh, this is about https://github.com/ros-perception/opencv_apps/commit/a0ddfdbee3b75d93081001d6cfcb53833b5f37f3.
I mean I just did `git rebase origin/master`.

> could you make better images which wen can used in wiki.ros.org?

OK.

**Comment by furushchev** [comment]
@k-okada Wiki is now updated. http://wiki.ros.org/opencv_apps

**Comment by k-okada** [comment]
thank you

**Pull request merged**